examples/lock-app/k32w/main/AppTask.cpp

```diff
@@ -394,48 +394,18 @@ void AppTask::LockActionEventHandler(AppEvent * aEvent)
 
 void AppTask::ThreadStart()
 {
-    chip::DeviceLayer::Internal::DeviceNetworkInfo networkInfo;
-
-    memset(networkInfo.ThreadNetworkName, 0, chip::DeviceLayer::Internal::kMaxThreadNetworkNameLength + 1);
-    memcpy(networkInfo.ThreadNetworkName, "OpenThread", 10);
-
-    networkInfo.ThreadExtendedPANId[0] = 0xde;
-    networkInfo.ThreadExtendedPANId[1] = 0xad;
-    networkInfo.ThreadExtendedPANId[2] = 0x00;
-    networkInfo.ThreadExtendedPANId[3] = 0xbe;
-    networkInfo.ThreadExtendedPANId[4] = 0xef;
-    networkInfo.ThreadExtendedPANId[5] = 0x00;
-    networkInfo.ThreadExtendedPANId[6] = 0xca;
-    networkInfo.ThreadExtendedPANId[7] = 0xfe;
-
-    networkInfo.ThreadMasterKey[0]  = 0x00;
-    networkInfo.ThreadMasterKey[1]  = 0x11;
-    networkInfo.ThreadMasterKey[2]  = 0x22;
-    networkInfo.ThreadMasterKey[3]  = 0x33;
-    networkInfo.ThreadMasterKey[4]  = 0x44;
-    networkInfo.ThreadMasterKey[5]  = 0x55;
-    networkInfo.ThreadMasterKey[6]  = 0x66;
-    networkInfo.ThreadMasterKey[7]  = 0x77;
-    networkInfo.ThreadMasterKey[8]  = 0x88;
-    networkInfo.ThreadMasterKey[9]  = 0x99;
-    networkInfo.ThreadMasterKey[10] = 0xAA;
-    networkInfo.ThreadMasterKey[11] = 0xBB;
-    networkInfo.ThreadMasterKey[12] = 0xCC;
-    networkInfo.ThreadMasterKey[13] = 0xDD;
-    networkInfo.ThreadMasterKey[14] = 0xEE;
-    networkInfo.ThreadMasterKey[15] = 0xFF;
-
-    networkInfo.ThreadPANId   = 0xabcd;
-    networkInfo.ThreadChannel = 15;
-
-    networkInfo.FieldPresent.ThreadExtendedPANId = true;
-    networkInfo.FieldPresent.ThreadMeshPrefix    = false;
-    networkInfo.FieldPresent.ThreadPSKc          = false;
-    networkInfo.NetworkId                        = 0;
-    networkInfo.FieldPresent.NetworkId           = true;
+    chip::DeviceLayer::Internal::ThreadOperationalDataset dataset;
+    uint8_t xpanid[]    = { 0xde, 0xad, 0x00, 0xbe, 0xef, 0x00, 0xca, 0xfe };
+    uint8_t masterkey[] = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
+
+    dataset.SetNetworkName("OpenThread");
+    dataset.SetExtendedPanId(xpanid);
+    dataset.SetMasterKey(masterkey);
+    dataset.SetPanId(0xabcd);
```

Comment:
nit: could we add constants like `kPanId` and `kThreadChannel` ? wondering if the constants could be grouped together with the xpanid and masterkey.

Same for below.